src/Nncase.Core/IR/Expr.cs

```diff
@@ -85,20 +86,38 @@ public IRType CheckedType
     /// <summary>
     /// Gets checked shape.
     /// </summary>
-    public Shape CheckedShape => CheckedType switch
+    public Shape CheckedShape
     {
-        TensorType type => type.Shape,
-        _ => throw new InvalidOperationException("Only The Expr Have CheckedType Can Get It's Shape"),
-    };
+        get
+        {
+            switch (CheckedType)
+            {
+                case TensorType type:
+                    return type.Shape;
+                default:
+                    DumpScope.Current.DumpIR(this, "CheckedShapeError");
```

Comment:
Wrap the dump with a flag, it's not necessary to dump the expr in every situation.

Reply:
changed.